doc/fuzzing.md

```diff
@@ -113,6 +113,15 @@ the qa-assets repo
 Patience is useful; even with improved throughput, libFuzzer may need days and
 10s of millions of executions to reach deep/hard targets.
 
+### Using the MemorySanitizer (MSan)
+
+MSan [requires](https://clang.llvm.org/docs/MemorySanitizer.html#handling-external-code)
+that all linked code be instrumented. It's necessary to compile a custom libc++ to run a Bitcoin
+Core fuzz target with MSan. The exact steps to achieve this may vary but involve compiling `clang`
```

Comment:
Is it necessary to compile `clang` from source to build an instrumentalized libc++? I figured probably not, but maybe building a libc++ of a given version requires the `clang` binary for this very version? Or maybe not required but recommended? Anyhow i kept it here as that's what the MSan CI job does.

Reply:
> Is it necessary to compile clang from source to build an instrumentalized libc++?

No I don't think so but the simplest/foolproof way to do it is to just clone llvm and build everything from scratch yourself. Building clang yourself ensures you avoid any version related issues (like you mentioned).